packages/color-scale-block/src/ColorScaleBlock.tsx

```diff
@@ -99,368 +79,152 @@ export const ColorScaleBlock: FC<BlockProps> = ({ appBridge }) => {
+                                                    const startWidth = displayableItems[idx].width;
+                                                    const nextWidth = displayableItems[idx + 1].width;
+
+                                                    const onMouseMove = (e: MouseEvent) => {
+                                                        const diff = e.clientX - startX;
+                                                        const percentage =
+                                                            totalWidth /
+                                                            (colorScaleBlockRef.current?.clientWidth ?? 800);
+                                                        const newWidth = startWidth + diff * percentage;
+                                                        const newNextWidth = nextWidth - diff * percentage;
+                                                        if (
+                                                            newWidth > MINIMUM_COLOR_WIDTH &&
+                                                            newNextWidth > MINIMUM_COLOR_WIDTH
+                                                        ) {
+                                                            setDisplayableItems((prev) => {
+                                                                const newColors = [...prev];
+                                                                newColors[idx].width = newWidth;
+                                                                newColors[idx + 1].width = newNextWidth;
+                                                                return newColors;
+                                                            });
+                                                        }
+                                                    };
+                                                    const onMouseUp = () => {
+                                                        window.removeEventListener('mousemove', onMouseMove);
+                                                        window.removeEventListener('mouseup', onMouseUp);
+                                                        setBlockSettings({ colorInput: displayableItems });
+                                                    };
+                                                    window.addEventListener('mousemove', onMouseMove);
+                                                    window.addEventListener('mouseup', onMouseUp);
+                                                }}
+                                            >
+                                                <div
+                                                    data-test-id="drag-handle"
+                                                    className="tw-bg-black-20 tw-rounded-full tw-w-2 tw-h-6"
+                                                />
+                                            </div>
```

Comment:
These lines are more or less the key of this open PR. For someone who takes it over: this resizing function is straight forward and nice to have for this block. 

cc @ragi96